projects/packages/account-protection/src/class-password-detection.php

```diff
@@ -41,13 +42,17 @@ public function __construct( ?Email_Service $email_service = null, ?Validation_S
 	/**
 	 * Check if the password is safe after login.
 	 *
-	 * @param \WP_User|\WP_Error $user The user or error object.
-	 * @param string             $password The password.
+	 * @param \WP_User|\WP_Error|null $user The user or error object, or null.
+	 * @param string                  $password The password.
 	 *
-	 * @return \WP_User|\WP_Error The user object.
+	 * @return \WP_User|\WP_Error|null The user object, error object, or null.
 	 */
 	public function login_form_password_detection( $user, string $password ) {
-		if ( is_wp_error( $user ) || ! $this->user_requires_protection( $user, $password ) ) {
+		if ( is_wp_error( $user ) || $user === null || ! ( $user instanceof WP_User ) ) {
```

Comment:
This check can be simplified:

```suggestion
		if ( is_wp_error( $user ) || ! ( $user instanceof WP_User ) ) {
```

Reply:
I'd suggest some adding a comment that documents why we need this extra check (third-party plugins have issues).

Reply:
Thanks, I just removed the explicit null check and added a comment :)